### ISSUE TITLE
Tab Index Broken

### ISSUE THREAD

**Issue by indolering** [PolymerElements/iron-list#151]
Unable to navigate between records because the tab index becomes broken when recycling elements. 

A `getNext(element|id)` and `getPrevious(element|id)` function would help with navigation between elements.  You could even have an `iron-list-item` behavior that injects  properties into the list item (such as first, last, next, previous) and triggers resize events.

**Comment by indolering** [comment]
Related to #141.

**Comment by tgsergeant** [comment]
This is an issue we're seeing in Chrome's history UI.

To expand a bit more on this, [this demo](https://elements.polymer-project.org/bower_components/iron-list/demo/collapse.html) exhibits the problem:

* Hit tab to focus the toolbar elements
* Hit tab 30ish more times to start cycling through the iron-list elements
* Somewhere in there (at the point where physical elements start to be recycled), focus will go back to the toolbar before continuing through the rest of the iron-list elements

Additionally, if I hit tab from the bottom of the list, an element partway through the list will end up focused.

I suspect you would be able to work around this by setting tabindex based on the list `{{index}}`, but I haven't tried that myself.

(cc @tjsavage)

**Comment by tjsavage** [comment]
cc @blasten with @tgsergeant's repro

**Comment by blasten** [comment]
@tgsergeant  Please take a look at https://github.com/PolymerElements/iron-list/pull/179.
> iron-list can manage the items' focus state automatically. For keyboard navigation,
users can use the up and down keys to move to the previous and next items in the list.
To enable focus management, you can bind to the tabIndex property. e.g.

:)
